File: 백준/Gold/5430. AC/AC.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');
const T = Number(input.shift())
let idx = 0;

for (let t = 0; t < T; t++) {
    const p = input[idx++];
    const n = Number(input[idx++]);
    const arr = input[idx++];
    
    let dq;
    if (arr === '[]') {
        dq = [];
    } else {
        dq = arr.slice(1, -1).split(',').map(Number);
    }
    
    let isReversed = false;
    let error = false;
    
    for (let cmd of p) {
        if (cmd === 'R') {
            isReversed = !isReversed;
        } else if (cmd === 'D') {
            if (dq.length === 0) {
                error = true;
                break;
            }
            
            if (isReversed) {
                dq.pop();
            } else {
                dq.shift();
            }
        }
    }
    
    if (error) {
        console.log('error');
    } else {
        if (isReversed) {
            dq = dq.reverse();
        }
        console.log('[' + dq.join(',') + ']');
    }
}

File: 백준/Gold/5430. AC/AC.py
import sys
from collections import deque
input = sys.stdin.readline

T = int(input())
for _ in range(T):
    p = input().strip()
    n = int(input())
    arr = input().strip()
    
    if arr == '[]':
        dq = deque()
    else:
        dq = deque(map(int, arr[1:-1].split(',')))
        
    is_reversed = False
    error = False
    
    for cmd in p:
        if cmd == 'R':
            is_reversed = not is_reversed
        elif cmd == 'D':
            if not dq:
                error = True
                break
                
            if is_reversed:
                dq.pop()
            else:
                dq.popleft()
                
    if error:
        print('error')
    else:
        if is_reversed:
            dq = list(reversed(dq))
        print('[' + ','.join(map(str, dq)) + ']')